### PR TITLE
Don't cast boolean filter values to booleans

### PR DESCRIPTION
**Pull request by proteusvacuum**
@NoahCarnahan 

Requires: https://github.com/dimagi/dimagi-utils/pull/411

### CODE REVIEW

File: corehq/apps/userreports/reports/view.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def query_dict_to_dict(query_dict, domain):
     :param domain:
     :return: a dict
     """
-    request_dict = json_request(query_dict)
+    request_dict = json_request(query_dict, booleans_as_strings=True)
```
Comment (NoahCarnahan) [inline corehq/apps/userreports/reports/view.py]: Does this work when filtering both string and boolean typed columns? Is it worth writing some tests?

Comment (proteusvacuum) [inline corehq/apps/userreports/reports/view.py]: I think yes.
Booleans in UCR are modeled as `0` or `1`, so filters actually need to be of type `integer`. There is no `boolean` filter type.

**Comment by proteusvacuum** [comment]
✅

File: corehq/apps/userreports/reports/view.py (hunk 1 above)
Comment (sravfeyn) [inline corehq/apps/userreports/reports/view.py]: I changed this in favour of https://github.com/dimagi/commcare-hq/pull/16575/ (that does more explicit casting and has tests)

**Pull request merged**